### ISSUE TITLE
Request to add commands.bin.gz loading/saving to investigate the Switch version script changes.

### ISSUE THREAD

**Issue by mc68k** [MattFiler/OpenCAGE#62]
Feature request to allow OpenCAGE to load/save the Switch format of commands.pak.  The script files in this version are stored in gzip archives.  Extracting the archive yields a commands.bin that OpenCAGE doesn't recognize.

While playing, I had noticed that some script issues have been fixed.  Example, locking the door at the top of the stairs on the way to pick up the revolver in order to skip the hack when returning.

I can provide a plethora of commands.bin.gz files for testing.

**Comment by MattFiler** [comment]
A:I's scripting "commands" files have two formats:
* BIN - this is used in 64-bit versions of the game
* PAK - this is used in the 32-bit versions of the game

Windows (Steam/EGS) is 32-bit, along with Xbox 360 & PS3. Everything else is 64-bit.
I opted to investigate the PAK version rather than BIN as OpenCAGE primarily targets the EGS/Steam versions of the game.

The BIN.GZ variant is a version that Feral introduced for the Switch and iOS/Android ports which is a compressed version of the 64-bit BIN if I remember correctly.

Adding support for BIN scripts is on the roadmap after getting further with PAK support, and I have already done a lot of investigating of the format of that file, but not implemented any of it yet into the editor as it adds some extra complexity. Once it has been implemented, supporting the BIN.GZ variant should be very simple.

Will keep this issue open to highlight the above info!

**Comment by mc68k** [comment]
I had some questions about the ms store/gamepass version, so I did some digging.  The exe is compiled as 32 bit.  The game contains both commands.pak and commands.bin.  So I tried installing some mods, none of them worked.  Seems this version doesn't use the pak format despite being 32 bit.

I confirmed this by deleting the level .pak files and starting a new game.  No issues at all.  Then I restored the .pak files and deleted the commands.bin files.  This caused the game to CTD.

I would appear that despite being a 32 bit exe, it's using the 64 bit commands.bin files.

Also, there is an additional folder called entity in the env directory, that contains behavior_tree.db and material_mappings.pak

**Comment by MattFiler** [comment]
Yep, the Windows Store version of the game is 64-bit, as I say it's only the Steam/EGS versions on PC that are 32-bit.

**Comment by MattFiler** [comment]
Yep, the Windows Store version of the game is "next-gen" and uses the 64-bit BIN version, as I say it's only Steam/EGS that use the 32-bit PAK version.

**Comment by MattFiler** [comment]
After all this time, I've finally started looking into this! I've got the BIN commands loading in, just need to write them back out (and fix up some issues with CAGEAnimations).

**Comment by mc68k** [comment]
Will a format conversion be a possibility? E.G. make edits on a 32-bit version and save those as 64-bit (or both formats)?

**Comment by MattFiler** [comment]
Yep! They'll be cross compatible, so previous mods will be able to be upgraded to 64-bit :)

**Comment by MattFiler** [comment]
This is now live if you opt in to the staging branch via the OpenCAGE settings (or Steam settings, if using it there) - just ensure "Save Commands PAK and BIN" is checked in the script editor's options, and you'll get a PAK and BIN saved out 👍
